packages/@aws-cdk/aws-iam/lib/policy-document.ts

```diff
@@ -43,16 +63,21 @@ export class PolicyDocument implements cdk.IResolvable {
   public readonly creationStack: string[];
   private readonly statements = new Array<PolicyStatement>();
   private readonly autoAssignSids: boolean;
+  private readonly minimize?: boolean;
 
   constructor(props: PolicyDocumentProps = {}) {
     this.creationStack = cdk.captureStackTrace();
     this.autoAssignSids = !!props.assignSids;
+    this.minimize = props.minimize;
 
     this.addStatements(...props.statements || []);
   }
 
   public resolve(context: cdk.IResolveContext): any {
-    context.registerPostProcessor(new RemoveDuplicateStatements(this.autoAssignSids));
+    context.registerPostProcessor(new PostProcessPolicyDocument(
+      this.autoAssignSids,
+      this.minimize ?? cdk.FeatureFlags.of(context.scope).isEnabled(cxapi.IAM_MINIMIZE_POLICIES) ?? false,
```

Comment:
I'm not sure what you mean here?

Reply:
I mean you are applying the default value of the `minimize` prop here... But I'd rather have `this.minimize = props.minimize ?? cdk.FeatureFlags.of(context.scope).isEnabled(...)` done in the constructor for `PolicyDocument`.

Of course now that I visualize the actual changes that this takes, I notice that you could not get the `FeatureFlags` in the constructor since you wouldn't have a `context` there... and so you can ignore this comment... 🤣